dao/chat.go

```diff
@@ -199,3 +180,30 @@ func (d chatDao) CloseActivePoll(streamID uint) error {
 func (d chatDao) GetChatsByUser(userID uint) (chats []model.Chat, err error) {
 	return chats, d.db.Find(&chats, "user_id = ?", userID).Error
 }
+
+// GetChat returns a chat message with the given id, uses the userId to normalize the chat.
+func (d chatDao) GetChat(id uint, userID uint) (*model.Chat, error) {
+	var chat model.Chat
+
+	err := d.db.Preload("Replies").Preload("UserLikes").Preload("AddressedToUsers").Find(&chat, "id = ?", id).Error
+	if err != nil {
+		return &chat, err
+	}
+
+	normalizeChat(&chat, userID)
+	return &chat, nil
+}
+
+func normalizeChat(chat *model.Chat, userID uint) {
```

Comment:
Let's call this something like `prepareChat` and add a descriptive comment what it does to clear things up.

```suggestion
// prepareChat sets Liked to true if the user with userID liked the message and and adds the ids of the addressed users to it for further usage in the fronted. 
func prepareChat(chat *model.Chat, userID uint) {
```